pandas/io/excel/_openpyxl.py

```diff
@@ -451,3 +462,277 @@ def write_cells(self, cells, sheet_name=None, startrow=0, startcol=0,
                             xcell = wks.cell(column=col, row=row)
                             for k, v in style_kwargs.items():
                                 setattr(xcell, k, v)
+
+
+class _OpenpyxlReader(_BaseExcelReader):
+
+    def __init__(self, filepath_or_buffer):
+        """Reader using openpyxl engine.
+
+        Parameters
+        ----------
+        filepath_or_buffer : string, path object or Workbook
+            Object to be parsed.
+        """
+        err_msg = "Install openpyxl >= 2.4.0 for Excel with Openpyxl support"
+
+        try:
+            import openpyxl
+        except ImportError:
+            raise ImportError(err_msg)
+        else:
+            if openpyxl.__version__ < LooseVersion('2.4.0'):
+                raise ImportError(err_msg +
+                                  ". Current version " + openpyxl.__version__)
+        super().__init__(filepath_or_buffer)
+
+    @property
+    def _workbook_class(self):
+        from openpyxl import Workbook
```

Comment:
Move this to top level as well

Reply:
Are you sure about this? I followed the structure of /`pandas/io/excel/_xlrd.py`, and there all imports of the engine are delayed until the function is called. Wouldn't top level imports cause import errors if `openpyxl` is not installed?

Reply:
Yea you are correct so keep this where it is. There is a new helper method in panda.common._optional you should use to simplify though (new as of yesterday)